modules/end-user-guide/pages/adding-a-vscode-extension.adoc

```diff
@@ -0,0 +1,163 @@
+You can add extra parameters to a plug-in by modifying the `.che/che-theia-plugins.yaml` file. These modifications include:
+* Defining the plug-ins for workspace installation.
+* Changing the default memory limit.
+* Overriding default preferences.
+
+
+=== Defining the plug-ins for workspace installation
+Define the plug-ins to be installed in the workspace.
+
+.Prerequisites
+. You have the `.che/che-theia-plugins.yaml` file in the root of the GitHub repository.
+
+.Procedure
+. Add the id of the plug-in to the  `.che/che-theia-plugins.yaml` file. Use the `/` sign to separate the publisher and plug-in name:
++
+*EXAMPLE*
+[source,yaml,subs="+quotes"]
+----
+- id: redhat/java  
+----
+
+=== Changing the default memory limit
+
+Override container settings such as the memory limit.
+
+.Prerequisites
+. You have the `.che/che-theia-plugins.yaml` file in the root of the GitHub repository.
+
+.Procedure
+. Add an `override` section in the `.che/che-theia-plugins.yaml` file under the `id` of the plug-in.
+. Specify the memory limit for the extension. In the following example, {prod-short} automatically installs the Red Hat Visual Studio Code Java extension in the {prod-short} workspace and increases the memory of the workspace by two gigabytes:
++
+*EXAMPLE*
+[source,yaml,subs="+quotes"]
+----
+
+  - id: redhat/java  
+    override:  
+      sidecar:  
+        memoryLimit: 2Gi
+----
+
+=== Overriding default preferences
+
+Override the default preferences of the Visual Studio Code extension for the workspace.
+
+.Prerequisites
+. You have the `.che/che-theia-plugins.yaml` file in the root of the GitHub repository.
+
+.Procedure
+. Add an `override` section in the `.che/che-theia-plugins.yaml` file under the `id` of the extension. 
+. Specify the preferences in the `Preferences` section. In the following example, {prod-short} automatically installs Red Hat Visual Studio Code Java extension in the {prod-short} workspace and sets the `java.server.launchMode` preference to `LightWeight`.
++
+*EXAMPLE*
+[source,yaml,subs="+quotes"]
+----
+  - id: redhat/java  
+    override:  
+      preferences:
+        java.server.launchMode: LightWeight
+----
+
+*NOTE* You can also define the preferences in the `.vscode/settings.json` file, either by changing the preferences in the UI of your IDE or by manually adding them to the `.vscode/settings.json` file:
+[source,json,subs="+quotes"]
+----
+{
+"my.preferences": "my-value"
+}
+----
+
+[id="visual-studio-code-extensions-in-devfile"]
+== Defining Visual Studio Code extension attributes in the devfile
+
+In case it's not possible to add extra files in the GitHub repository, you can define some of the plug-in or extension attributes by inlining them in the devfile. You can use this procedure with both `.vscode/extensions.json` and `.che/che-theia-plugins.yaml` file contents.
+
+=== Inlining `.vscode/extensions.json` file
+Use `.vscode/extensions.json` file contents to inline extension attributes in the devfile.
```

Comment:
Part of the issue with using the adjective, "inline," as a verb is that when you use it next to a noun, as shown here, the reader might experience some semantic noise:
raising the question, "Is a verb missing between "to" and "inline extension attributes?" 
Perhaps subsitute it with "add" or "insert?" WDYT?